request.go

```diff
@@ -254,3 +268,20 @@ func NewRequestWithContext(ctx context.Context, method, url string, body interfa
 
 	return &Request{httpReq, urlx, Metrics{}, nil}, nil
 }
+
+func updateScheme(u *url.URL) {
```

Comment:
@Mzack9999  , I think keeping this here would be more meaningful . since updating `req.URL` on fly is not recommended we should avoid and if we need we can do it using retryablehttp-go.

adding it to `urlutil.URL` would cause some confusion since this is for specific case and only happens with http.Request when URL is modified